go/vt/vtgate/engine/dml_with_input.go

```diff
@@ -58,6 +58,9 @@ func (dml *DMLWithInput) Inputs() ([]Primitive, []map[string]any) {
 
 // TryExecute performs a non-streaming exec.
 func (dml *DMLWithInput) TryExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, _ bool) (*sqltypes.Result, error) {
+	vcursor.Session().SetInDMLExecution(true)
```

Comment:
shouldn't we do this in the executor?

Reply:
This keeps it simple. We know for sure that we are executing a DML which has input.